### DISCUSSION TITLE
Add support for Temporary Chats only

### DISCUSSION THREAD

**Discussion by gaby** [open-webui/open-webui#5793] [Ideas]
Currently there's no way to disable chat history, there used to be a toggle but it was removed.

As an admin with a system of multiple users using Open-WebUI we want to be able to only allow Temporary Chats, basically the same approach OpenAI has with "https://chat.openai.com" when not logged in.

Once the user refresh or closes the browser the chat is gone.

**Comment by Minocula** [comment]
We use the open-webui docker container. Cool would be to have an enviroment setting for " Temporary_Chat_only=true". Thx

**Reply by gaby** [reply]
That's exactly what I'm looking for.

**Reply by liv-io** [reply]
I am also looking for this option +1

**Reply by tjbck** [reply]
It's a url search param but you might find this useful: `?temporary-chat=true`

**Reply by gaby** [reply]
@tjbck As an admin I don't want to allow chat history. So temporary chat seems to be the option?

There used to be a "Enable Chat History" toggle in the admin page but i cant find it anymore.

**Reply by gaby** [reply]
@tjbck Thoughts?

**Reply by ruizcrp** [reply]
+1

**Comment by DmitriyAlergant** [comment]
OpenWebUI's Temporary Chats are very limited -  you can't switch between more then one currently active Temporary Chat - unless in two separate browser tabs.  One wrong click on the New Chat button, or even refreshing a browser's tab - and the current temporary chat is gone - potentially with some work that went into it. This is a very limited user experience.

To protect potentially sensitive data from being persisted server-side, you can consider a backend solution periodically cleaning the Chat table in the database (SQLite or Postgre) based on logic you define - something like 4 to 24 hours since last message in the chat, or something like that.

**Reply by gaby** [reply]
That's the way it should be, similar to what `OpenAI` does with Free ChatGPT accounts (without an account).

**Reply by DmitriyAlergant** [reply]
That's the way what OpenAI does with Free ChatGPT accounts (without an account) - but it does not have to be what admins are forcing upon users. Not a good thing to force in terms of an experience.   OpenAI users are welcome to sign up and fix their experience.

As a dirty solution, you can spread the knowledge (of Temporary Chats feature), the direct access URL (via a query parameter)  - and also tamper with the backend database to prevent writes to the Chats table, like by adding a constraint to it.

**Comment by lowlyocean** [comment]
+1 for having an option to be able to have only temporary chats. For some people, there's little benefit in storing the chat history and it just takes up storage space unnecessarily and creates a liability from privacy/security perspective

**Comment by flefevre** [comment]
I am totaly aligned with at least an environmental variable for the server that impose temporary_chat=true (and by default false)
and it could be usefull to have also a way to start also chat without login.

**Comment by optimist555** [comment]
It's absolutely incredible that this feature is not just a simple setting. As an admin, why do I want to be able to see my users chats? Incredible privacy nightmare and a GDPR violation from my part, without any simple way to disable. I absolutely do not want or be able to see my users chats. At least add a setting to remove the ability for admin to see users chats.

**Reply by DmitriyAlergant** [reply]
Thris setting exists: ENABLE_ADMIN_CHAT_ACCESS=false

On the env var level (not set visually in admin UI) - makes it harder to re-enable

**Reply by optimist555** [reply]
My mistake! I will set it right away. Thanks!

**Comment by gaby** [comment]
I believe this is fixed by v0.6.0

**Reply by liv-io** [reply]
Thanks, the "Enforce Temporary Chat" option works indeed. No more pointless chat history :tada:
